### ISSUE TITLE
Update releases here on GitHub

### ISSUE THREAD

**Issue by Eagle3386** [sunjw/jstoolnpp#55]
@sunjw, I'd like to suggest an update to your [releases](https://github.com/sunjw/jstoolnpp/releases) section of jstoolnpp's page here on GitHub.
This way, we all get kind of an "one-stop experience" instead of "code on X, download on Y". 😃

**Comment by sunjw** [comment]
I'm afraid not until GitHub gives more statics on releases.

**Comment by Eagle3386** [comment]
What do you mean by stating "more statics on the releases tab"? 🤔

**Comment by sunjw** [comment]
@Eagle3386 sf.net gives lots of download statics. But GitHub releases tab is REALLY just a page for download. I'd like to know where my users from, how many users updated, how they download (from plugin manager or manually), which GitHub cannot tell me. And another problem is that downloading file from GitHub is difficult here in China, where about 15% downloads come from. So, move download to GitHub here, has no real benifit to me, I think. sf.net works well up to now. But things will change from time to time, let's see if GitHub will work better than sf.net later.

**Comment by Eagle3386** [comment]
@sunjw, sorry for the delay in replying.

While I understand your point, instead of considering it a drawback, I encourage you to see it as an advantage, because it increases your users privacy. Besides, how does it matter if installed via GH, SF, N++'s old or even N++'s coming plugin manager?

After SF treated users so bad, it earned the following distrust all by its own. Luckily, at the same time GH established itself as **the** platform for (open source) code sharing and even for file distribution.

Of course, having 15% of your users dealing with download problems due to "government issues" is a valid point. Therefore I'd like to suggest that instead of replacing SF, just use GH's release page as an additional download source. This way, you're providing my suggested "one-stop experience" for those coming here for it while still allowing those 15% to continue to download via alternative sources.

**Comment by sunjw** [comment]
Let's say: "**Things will change from time to time.**" I just leave this issue here, and maybe someday, I will close it.

**Comment by chcg** [comment]
See #45, with appveyor CI you could automate the GH release deployment just by creating git tags.

**Comment by supremehover** [comment]
In China, I can download files from Github. There is no GFW on the GitHub.
But, I cannot access the pages of SourceForge.net, there is a real "government issues" here.
We are all Chinese, why do you want us to access mirror site of SourceForge.net to download files? You know, it's boring.

**Comment by Eagle3386** [comment]
First of all, GFW has nothing to do with GitHub's builtin release feature for each repo.
Secondly, nobody mentioned SourceForge.net before you. It's not helping mixing this this into the discussion either.
Lastly, stating "it's boring" isn't a serious argument and despite your statement, *not* everybody is Chinese - I don't care for your government as it has serious issues with democracy and treats people sometimes like shit.

However, what I do care about is freedom of choice and privacy - neither is fulfilled by limiting access to releases of JSTool to an GH-external site and tracking how, where, etc. users are retrieving updates.
I'm almost convinced to fork this project and set up automatic releases via GH's builtin release feature.

**Comment by sunjw** [comment]
@Eagle3386 I can see that you miss understand @supremehover 's point. @supremehover is supporting your point.
I think what really important is that JSTool is still in active development and keeps bring value to users. I'm busy on writing JSTool for Visual Studio Code, recently. VSC is a growing platform and a great editor, with its own extension market (hosted by MS). I do think it is the "one-stop experience", and hope there will be no such GH/SF issue after JSTool succeed/failed on VSC platform.
Still, let's say: "Things will change from time to time."

**Comment by Eagle3386** [comment]
Apologies for misunderstanding you in the first place, @supremehover.

@sunjw: I see. Deleted my comment accordingly.
Since last weekend, I've switched to VSCode, too - so, I perfectly understand the effort you'r putting into that. Still, it doesn't make things easier regarding Notepad++..

**Comment by NudeSpoons** [comment]
@sunjw 
When you refer to "statics" on SourceForge, do you mean "statistics" or "metrics"?.

**Comment by CollinChaffin** [comment]
You mean like this?  If you're simply looking for download COUNTS, the API is there and easy to access:

![wGOboFAYGm](https://user-images.githubusercontent.com/9301483/63225243-4dbb8900-c193-11e9-9f3e-24c3f6eaddd2.png)

**Comment by Eagle3386** [comment]
Issue closed as `completed`, but no GitHub release?
Good luck, I'm off with other N++ JSON tools now.

**Comment by sunjw** [comment]
No, there will be no GitHub release. There will be some changes later.

**Comment by Eagle3386** [comment]
Yet another reason to skip this tool, thanks.
